packages/gatsby-theme-docs/src/components/search-input.js

```diff
@@ -77,6 +77,7 @@ const SearchInput = React.forwardRef((props, ref) => {
   const handleBlur = () => {
     if (props.isDisabled) return;
     setIsActive(false);
+    props.onClose();
```

Comment:
Unrelated to the original issue, but when using tab navigation for the search input, when the input blurs the search dialog should close as it's not the focused element anymore.